meow/frontend/src/components/EditPost/index.js

```diff
@@ -149,11 +150,21 @@ class EditPost extends React.Component {
       });
   };
 
+  popSectionerror = () => {
+    notification.open({
+      message: "Send Failed :(",
+      description: "No section was selected",
+      icon: <Icon type="close-circle" style={{ color: "#FF0000" }} />
+    });
+  };
+
   sendNow = () => {
     const { postId } = this.props.match.params;
 
     this.savePostPromise(postId).then(data => {
-      if (data) {
+      ++this.state.version_number;
```

Comment:
With state updates for React, you should almost always use `this.setState()`. The one exception is in the constructor or when you initialize the state like on line 28.

`this.setState(newState)` will set the current state to be `newState`. However, often times we only want to change 1 key of the state (in this case, we want to change `version_number`. So in order to only change one and leave the rest the same, you can do this:
```
this.setState({
   ...this.state, // this will copy all the old state into this new object your making,
   thing_you_want_to_change: newValue // this will modify the `thing_you_want_to_change` of this new object. If `...this.state` already had this key, it will override it and set it to `newValue` instead of whatever was in `this.state`
})
```

Reply:
now changed by setState